src/state/user.ts

```diff
@@ -28,10 +28,19 @@ export enum Deadlines {
   Custom = "CUSTOM",
 }
 
+export type ConfirmModalType = {
+  open: boolean
+  options?: {
+    modalTitle?: string
+    modalText?: string
+    onOK?: () => void
+    onCancel?: () => void
```

Comment:
I don't know if you can safely persist this to user state. The user state is written to local storage and is first serialized to json. The app state then deserializes from local storage to hydrate, so the function wouldn't work